### PR TITLE
Driving robot

### PR DESCRIPTION
**Pull request by Elised9**
The robot can now drive automatically or with a controller. Different motor types, a working drive base, and successful control of the robot with the xbox controller. We tested it on the robot, and we just need to figure out how to get the right side working. It works.

### CODE REVIEW

File: TitanRobotics2024/src/main/java/frc/robot/Auto/AutoMissionExecutor.java [hunk 1]
```diff
@@ -21,11 +21,13 @@ public void run(){
         });
     }
 
-    public void start() {
-        if (mThread != null) {
-            mThread.start();
+    public void start() 
+        {
```
Comment (thebuilder6) [inline TitanRobotics2024/src/main/java/frc/robot/Auto/AutoMissionExecutor.java:25]: Don't do this double tabbing thing, please

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/Control.java [hunk 2]
```diff
@@ -16,6 +19,14 @@ public static Control getInstance()
     public Control()
     {
         driveBase = DriveBase.getInstance();
+        driverController = DriverController.getInstance();
+    }
+
+    public void TeleopControl()
```
Comment (thebuilder6) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/Control.java:25]: refactor the function name to follow camelCase convention
 maybe rename `forward` and `turn` for increased readability

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriveBase.java [hunk 3]
```diff
@@ -2,71 +2,60 @@
...
+  private double rightPower;
 
-  private final ModifiedMotors motorFrontLeft;
-  private final ModifiedMotors motorRearLeft;
-  private final ModifiedMotors motorFrontRight;
-  private final ModifiedMotors motorRearRight;
+  private ModifiedMotors motorRearLeft;
+  private ModifiedMotors motorFrontRight;
+  private ModifiedMotors motorRearRight;
+  private ModifiedMotors motorFrontLeft;
```
Comment (thebuilder6) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriveBase.java:15]: 'rearLeftMotor' naming convention slightly prefered over 'motorRearLeft'

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriveBase.java [hunk 4]
```diff
@@ -2,71 +2,60 @@
...
-  {
-    // set the left side motors to "power"
+  public void setLeftSideMotors(double power) {
+    this.leftPower = power;
   }
 
   /**
    * Control Type: Left Stick controls speed; Right Stick controls direction (Used
    * in tank)
    */
```
Comment (thebuilder6) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriveBase.java:44]: Remove/replace inaccurate/old comments

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriveBase.java [hunk 5]
```diff
@@ -2,71 +2,60 @@
...
+    this.motorRearLeft   =  new ModifiedMotors(PortMap.REARLEFT.portNumber, "CANVictorSPX");
+    this.motorFrontRight =  new ModifiedMotors(PortMap.FRONTRIGHT.portNumber, "CANVictorSPX");
+    this.motorRearRight  =  new ModifiedMotors(PortMap.REARRIGHT.portNumber, "CANVictorSPX");
+
   }
 
-  public void setRightSideMotors(double power) 
-  {
-    // set the right side motors to "power"
+  public void setRightSideMotors(double power) {
```
Comment (thebuilder6) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriveBase.java:33]: Prefer refactorring function name to 'setRightPower()', shorter an more accurate discription

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriverController.java [hunk 6]
```diff
@@ -14,21 +20,43 @@ public static DriverController getInstance()
...
-        
+        this.xboxController = new XboxController(PortMap.XBOX_DRIVER_CONTROLLER.portNumber);       
     }
 
-    @Override
-    public void start() 
-    {
-        // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'start'");
+    public double getStick(ButtonMap stickAxis)
```
Comment (thebuilder6) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriverController.java:26]: Fix function formatting

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/ModifiedMotors.java [hunk 7]
```diff
@@ -1,39 +1,51 @@
...
         // TODO Auto-generated method stub
         throw new UnsupportedOperationException("Unimplemented method 'update'");
     }
-    
-     private int portNumber;
+
+    private int portNumber;
     private final MotorController motor;
-    
-    public ModifiedMotors(int portNumber)
```
Comment (thebuilder6) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/ModifiedMotors.java:25]: Maybe encapsulate the motor initialization logic into separate methods for improved readability and maintainability
``` 
public ModifiedMotors(int portNumber, String motorType) {
        this.portNumber = portNumber;
        MotorController motorTemporarily = null;
        switch (motorType) {
            case "PWMVictorSPX":
                motorTemporarily = initializePWMVictorSPX(portNumber);
                break;
            case "CANVictorSPX":
                motorTemporarily = initializeCANVictorSPX(portNumber);
                break;
            default:
                System.err.println("Error: motors not activated");
        }
        motor = motorTemporarily;
    }

    private MotorController initializePWMVictorSPX(int portNumber) {
        try {
            return new PWMVictorSPX(portNumber);
        } catch (Exception e) {
            System.err.println("Error: Port Not Activated " + portNumber);
            return null;
        }
    }

    private MotorController initializeCANVictorSPX(int portNumber) {
        try {
            return new WPI_VictorSPX(portNumber);
        } catch (Exception e) {
            System.err.println("Error: CANID Not Activated " + portNumber);
            return null;
        }
    }
```

Comment (thebuilder6) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/ModifiedMotors.java:25]: add a function comment that gives available options 
```
/**
 * Constructor for ModifiedMotors class.
 * Initializes the motor based on the specified motor type and port number.
 * 
 * @param portNumber the port number of the motor
 * @param motorType the type of the motor (e.g., "PWMVictorSPX" or "CANVictorSPX")
 */
 ```

**Review by thebuilder6 — CHANGES_REQUESTED** [review]
edited 
see below

File: TitanRobotics2024/src/main/java/frc/robot/Auto/AutoMissionExecutor.java [hunk 8]
```diff
@@ -21,11 +21,13 @@ public void run(){
...
+            if (mThread != null) 
+                {
+                    mThread.start();
+                }
         }
-    }
```
Comment (thebuilder6) [inline TitanRobotics2024/src/main/java/frc/robot/Auto/AutoMissionExecutor.java:28]: Don't do this double tabbing thing, please

File: TitanRobotics2024/src/main/java/frc/robot/ExternalLibraries/LimelightHelpers.java
Comment (thebuilder6) [inline TitanRobotics2024/src/main/java/frc/robot/ExternalLibraries/LimelightHelpers.java:1]: Remove limelight specific files and let them be added in own pull request when ready

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/AprilTagTargeting.java
Comment (thebuilder6) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/AprilTagTargeting.java:1]: Remove limelight specific files and let them be added in own pull request when ready

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/Control.java [hunk 9]
```diff
@@ -16,6 +19,14 @@ public static Control getInstance()
...
+    {
+        double forward = driverController.getStick(ButtonMap.XboxLEFTSTICKY);
+        double turn = driverController.getStick(ButtonMap.XboxRIGHTSTICKX);
+        driveBase.drive(forward, turn);
     }
```
Comment (thebuilder6) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/Control.java]: refactor the function name to follow camelCase convention
maybe rename `forward` and `turn` for increased readability

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriveBase.java [hunk 10]
```diff
@@ -2,71 +2,60 @@
...
+  private double rightPower;
 
-  private final ModifiedMotors motorFrontLeft;
-  private final ModifiedMotors motorRearLeft;
-  private final ModifiedMotors motorFrontRight;
-  private final ModifiedMotors motorRearRight;
+  private ModifiedMotors motorRearLeft;
+  private ModifiedMotors motorFrontRight;
+  private ModifiedMotors motorRearRight;
+  private ModifiedMotors motorFrontLeft;
```
Comment (thebuilder6) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriveBase.java]: `rearLeftMotor` naming convention slightly prefered over `motorRearLeft`

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriveBase.java [hunk 11]
```diff
@@ -2,71 +2,60 @@
...
+  public void setRightSideMotors(double power) {
+    this.rightPower = power;
   }
 
-  public void setLeftSideMotors(double power) 
-  {
-    // set the left side motors to "power"
+  public void setLeftSideMotors(double power) {
+    this.leftPower = power;
   }
```
Comment (thebuilder6) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriveBase.java]: Prefer refactoring function names to `setRightPower()` shorter and more accurate discription

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriveBase.java [hunk 12]
```diff
@@ -2,71 +2,60 @@
...
    * in tank)
    */
-  public void drive(double forward, double turn) 
-  {
+  public void drive(double forward, double turn) {
+    this.leftPower = (forward - (0.35 * turn));// invert using negetive for left side motors
+    this.rightPower = (forward + (0.35 * turn));
+  
     // put code here that drives the bot using the inputs "forward" and "turn"
   }
```
Comment (thebuilder6) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriveBase.java]: Remove/replace inaccurate/old comments

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriverController.java [hunk 13]
```diff
@@ -14,21 +20,43 @@ public static DriverController getInstance()
...
+        {
+            SmartDashboard.putString("ControllerError", "AxisNotFound");
+            return 0;
+        }
+        } 
+        else 
+        {
+        return 0;
+        }     
     }
```
Comment (thebuilder6) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriverController.java]: Fix function formatting

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/ModifiedMotors.java [hunk 14]
```diff
@@ -1,39 +1,51 @@
...
+                    System.err.println("Error: CANID Not Activated" + this.portNumber);
+                }
+                break;
+            default:
+            System.err.println("Error: motors not activated");
             motorTemporarily = null;
-            SmartDashboard.putNumber("Error: Port Not Activated", this.portNumber);
         }
         motor = motorTemporarily;
     }
```
Comment (thebuilder6) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/ModifiedMotors.java]: Maybe encapsulate the motor initialization logic into separate methods for improved readability and maintainability
``` 
public ModifiedMotors(int portNumber, String motorType) {
      this.portNumber = portNumber;
      MotorController motorTemporarily = null;
      switch (motorType) {
          case "PWMVictorSPX":
              motorTemporarily = initializePWMVictorSPX(portNumber);
              break;
          case "CANVictorSPX":
              motorTemporarily = initializeCANVictorSPX(portNumber);
              break;
          default:
              System.err.println("Error: motors not activated");
      }
      motor = motorTemporarily;
  }

  private MotorController initializePWMVictorSPX(int portNumber) {
      try {
          return new PWMVictorSPX(portNumber);
      } catch (Exception e) {
          System.err.println("Error: Port Not Activated " + portNumber);
          return null;
      }
  }

  private MotorController initializeCANVictorSPX(int portNumber) {
      try {
          return new WPI_VictorSPX(portNumber);
      } catch (Exception e) {
          System.err.println("Error: CANID Not Activated " + portNumber);
          return null;
      }
  }
```

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/ModifiedMotors.java [hunk 15]
```diff
@@ -1,39 +1,51 @@
...
-    
-     private int portNumber;
+
+    private int portNumber;
     private final MotorController motor;
-    
-    public ModifiedMotors(int portNumber) 
-    {
+
+    public ModifiedMotors(int portNumber, String motorType) {
```
Comment (thebuilder6) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/ModifiedMotors.java]: add a function comment that gives available options 
```
/**
 * Constructor for ModifiedMotors class.
 * Initializes the motor based on the specified motor type and port number.
 * 
 * @param portNumber the port number of the motor
 * @param motorType the type of the motor (e.g., "PWMVictorSPX" or "CANVictorSPX")
 */
 ```

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/Ramp.java
Comment (thebuilder6) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/Ramp.java:1]: Commit to own branch

**Review by thebuilder6 — CHANGES_REQUESTED** [review]
I added several comments on things that really should be changed 
and a few comments on stylistic things that could be changed

@Elised9 @dalittlet Also maybe review my review and see if this is how we want to do it, was i too picky? not picky enough? want more or less detail?

Duplication of review as I ended up wanting to change how I did comments

File: TitanRobotics2024/src/main/java/frc/robot/Auto/AutoMissionExecutor.java [hunk 16]
```diff
@@ -21,11 +21,13 @@ public void run(){
         });
     }
```
Comment (Elised9) [inline TitanRobotics2024/src/main/java/frc/robot/Auto/AutoMissionExecutor.java:29]: make sure the starting curly brackets are on the next line

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriverController.java [hunk 17]
```diff
@@ -14,21 +20,43 @@ public static DriverController getInstance()
...
 
     @Override
     public void update() 
     {
         // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'update'");
+        // throw new UnsupportedOperationException("Unimplemented method 'update'");
```
Comment (dalittlet) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriverController.java]: Remove commented out code

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/ModifiedMotors.java [hunk 18]
```diff
@@ -1,39 +1,51 @@
...
 
 public class ModifiedMotors implements Subsystem {
 
-
-    @Override
-    public void start() {
-        // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'start'");
-    }
+    // @Override
```
Comment (dalittlet) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/ModifiedMotors.java]: Remove commented out code

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/OperatorController.java [hunk 19]
```diff
@@ -17,18 +17,11 @@ public OperatorController()
...
-        // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'start'");
-    }
-
     @Override
     public void update() 
     {
         // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'update'");
+       // throw new UnsupportedOperationException("Unimplemented method 'update'");
```
Comment (dalittlet) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/OperatorController.java]: Remove commented out code

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/AprilTagTargeting.java
Comment (Elised9) [reply TitanRobotics2024/src/main/java/frc/robot/Subsystem/AprilTagTargeting.java:1]: Is it possible that they took the code from the limelight branch?

File: TitanRobotics2024/src/main/java/frc/robot/Auto/AutoMissionExecutor.java (hunk 16 above)
Comment (thebuilder6) [reply TitanRobotics2024/src/main/java/frc/robot/Auto/AutoMissionExecutor.java:29]: See issue #4

**Review by dalittlet — CHANGES_REQUESTED** [review]
Make sure to remove commented out code

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriverController.java (hunk 17 above)
Comment (thebuilder6) [reply TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriverController.java]: If they think they might use it later, what should they do?

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/ModifiedMotors.java [hunk 20]
```diff
@@ -1,39 +1,51 @@
...
+    // public void start() {
+    // // TODO Auto-generated method stub
+    // throw new UnsupportedOperationException("Unimplemented method 'start'");
+    // }
```
Comment (Elised9) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/ModifiedMotors.java:10]: get ride of commented out stuff

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/ModifiedMotors.java (hunk 15 above)
Comment (Elised9) [reply TitanRobotics2024/src/main/java/frc/robot/Subsystem/ModifiedMotors.java]: also, fix function start.

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/ModifiedMotors.java (hunk 14 above)
Comment (Elised9) [reply TitanRobotics2024/src/main/java/frc/robot/Subsystem/ModifiedMotors.java]: I would say put the catch line under the end brakets

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/Ramp.java [hunk 21]
```diff
@@ -0,0 +1,19 @@
...
+public class Ramp implements Subsystem
+{
+    private static Ramp instance = null;
+    
+    // @Override
+    // public void start()
+    // {
+
+    // }
+
```
Comment (Elised9) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/Ramp.java]: get rid of commented stuff

**Review by Elised9 — COMMENTED** [review]
I left a few comments. I agree with what thebuilder6 has commented.

File: TitanRobotics2024/src/main/java/frc/robot/Auto/AutoMissionExecutor.java [hunk 22]
```diff
@@ -21,11 +21,13 @@ public void run(){
...
-    public void start() {
-        if (mThread != null) {
-            mThread.start();
+    public void start() 
+        {
+            if (mThread != null) 
+                {
+                    mThread.start();
+                }
         }
```
Comment (thebuilder6) [inline TitanRobotics2024/src/main/java/frc/robot/Auto/AutoMissionExecutor.java]: Jacob removed my comment but i still don't like the double tabs

**Review by thebuilder6 — CHANGES_REQUESTED** [review]

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriveBase.java (hunk 12 above)
Comment (impossiblerocket) [reply TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriveBase.java]: resolved

**Review by thebuilder6 — APPROVED** [review]

File: TitanRobotics2024/src/main/java/frc/robot/Auto/AutoMissionExecutor.java [hunk 23]
```diff
@@ -3,17 +3,18 @@
...
+ * Mission.
  */
 public class AutoMissionExecutor {
     private MissionBase mAutoMission = null;
     private Thread mThread = null;
 
     public void setAutoMission(MissionBase new_auto_Mission) {
         mAutoMission = new_auto_Mission;
-        mThread = new Thread( new Runnable(){
+        mThread = new Thread(new Runnable() {
```
Comment (dalittlet) [inline TitanRobotics2024/src/main/java/frc/robot/Auto/AutoMissionExecutor.java]: coding standards

File: TitanRobotics2024/src/main/java/frc/robot/Auto/AutoMissionExecutor.java [hunk 24]
```diff
@@ -3,17 +3,18 @@
...
             @Override
-            public void run(){
+            public void run() {
```
Comment (dalittlet) [inline TitanRobotics2024/src/main/java/frc/robot/Auto/AutoMissionExecutor.java]: coding standards

File: TitanRobotics2024/src/main/java/frc/robot/Robot.java [hunk 25]
```diff
@@ -35,44 +38,65 @@ public class Robot extends TimedRobot {
...
    *
-   * <p>This runs after the mode specific periodic functions, but before LiveWindow and
+   * <p>
+   * This runs after the mode specific periodic functions, but before LiveWindow
+   * and
    * SmartDashboard integrated updating.
    */
   @Override
-  public void robotPeriodic() {}
+  public void robotPeriodic() {
```
Comment (dalittlet) [inline TitanRobotics2024/src/main/java/frc/robot/Robot.java]: coding standards

File: TitanRobotics2024/src/main/java/frc/robot/Robot.java [hunk 26]
```diff
@@ -92,35 +116,42 @@ public void autonomousPeriodic() {
 
   /** This function is called once when teleop is enabled. */
   @Override
-  public void teleopInit() {}
+  public void teleopInit() {
```
Comment (dalittlet) [inline TitanRobotics2024/src/main/java/frc/robot/Robot.java]: coding standards

File: TitanRobotics2024/src/main/java/frc/robot/Robot.java [hunk 27]
```diff
@@ -92,35 +116,42 @@ public void autonomousPeriodic() {
...
+  }
 
   /** This function is called periodically during operator control. */
   @Override
   public void teleopPeriodic() {
```
Comment (dalittlet) [inline TitanRobotics2024/src/main/java/frc/robot/Robot.java]: ALL OF THE STANDARDS

File: TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriveBase.java [hunk 28]
```diff
@@ -2,71 +2,53 @@
 
 import frc.robot.Data.PortMap;
 
-public class DriveBase implements Subsystem 
-{
+public class DriveBase implements Subsystem {
```
Comment (dalittlet) [inline TitanRobotics2024/src/main/java/frc/robot/Subsystem/DriveBase.java]: WHY DID YOU UNDO THE STANDARD

**Review by dalittlet — APPROVED** [review]
